Lib/test/test_capi/test_long.py

```diff
@@ -643,6 +643,51 @@ def test_long_getsign(self):
 
         # CRASHES getsign(NULL)
 
+    def test_long_ispositive(self):
+        # Test PyLong_IsPositive()
+        ispositive = _testcapi.pylong_ispositive
+        self.assertEqual(ispositive(1), 1)
+        self.assertEqual(ispositive(123), 1)
+        self.assertEqual(ispositive(-1), 0)
+        self.assertEqual(ispositive(0), 0)
+        self.assertEqual(ispositive(True), 1)
+        self.assertEqual(ispositive(False), 0)
+        self.assertEqual(ispositive(IntSubclass(-1)), 0)
+        self.assertRaises(TypeError, ispositive, 1.0)
+        self.assertRaises(TypeError, ispositive, Index(123))
```

Comment:
Speaking of error messages, use an `assertRaisesRegex` for this, since it's the only way that it can fail.

Reply:
We don't use assertRaisesRegex for others long API tests.

Reply:
I don't think consistency matters for tests. Is there any pain in using `assertRaisesRegex`?

Reply:
IMO, it doesn't have much sense, at least now.  Error messages aren't part of API and C-API functions like PyErr_ExceptionMatches() check only the exception type.

Reply:
My concern was that we're forcing these functions to *only* ever raise an "invalid type" error, per the documentation--we should test that nothing else is accidentally getting raised via `assertRaisesRegex`.

Reply:
I doubt it's possible.  We can test only finite (even not countable) inputs.

If you think that we should use more test cases - feel free to suggest concrete examples.  But I don't think that checking anything else than the exception type will be helpful.

Reply:
Really, we only need *one* case to make sure it's doing the right thing, but I agree that we don't need any more than that. Here, this should be fine:
```suggestion
        self.assertRaisesRegex(TypeError, 'expected int, got float', ispositive, 1.0)
        self.assertRaises(TypeError, ispositive, Index(123))
```

Reply:
What's wrong if next time we rephrase the exception message a bit?:)  E.g. "expected integer, got float".

Reply:
That's a breaking change. Users unfortunately rely on exception messages for things like this.

Reply:
Peter is correct that there already is more than one way to produce a `TypeError` here; checking for the correct `TypeError` makes sense. We do this multiple places in the test suite already. I agree that there is no need to check for the _exact_ phrase, though. A convenience `check(...)` function can be added to reduce boiler-plate test code.

Reply:
FTR, for this PR I think it is ok to use `assertRaises`; these are pretty basic C APIs.

Reply:
It's not a deal breaker here, but I would be much more comfortable if we had even just one case that checked the error message with `assertRaisesRegex`.